exporter/clickhouseexporter/internal/metrics_model.go

```diff
@@ -0,0 +1,463 @@
+	metricsMap := make(map[pmetric.MetricType]MetricsModel)
+	metricsMap[pmetric.MetricTypeGauge] = &GaugeMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertGaugeTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeSum] = &SumMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertSumTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeHistogram] = &HistogramMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertHistogramTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeExponentialHistogram] = &ExpHistogramMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertExpHistogramTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeSummary] = &SummaryMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertSummaryTableSQL, "'", "`"), tableName),
+	}
+	return metricsMap
+}
+
+func InjectMetaData(metricsMap map[pmetric.MetricType]MetricsModel, metaData *MetricsMetaData) {
+	for _, metrics := range metricsMap {
+		metrics.InjectMetaData(metaData)
+	}
+}
+
+func InsertMetrics(ctx context.Context, tx *sql.Tx, metricsMap map[pmetric.MetricType]MetricsModel, logger *zap.Logger) error {
+	var errs []error
+	for _, metrics := range metricsMap {
+		errs = append(errs, metrics.Insert(ctx, tx, logger))
+	}
+	return multierr.Combine(errs...)
+}
+
+func convertExemplars(exemplars pmetric.ExemplarSlice) (clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet) {
+	var (
+		attrs       clickhouse.ArraySet
+		times       clickhouse.ArraySet
+		floatValues clickhouse.ArraySet
+		intValues   clickhouse.ArraySet
+		traceIDs    clickhouse.ArraySet
+		spanIDs     clickhouse.ArraySet
+	)
+	for i := 0; i < exemplars.Len(); i++ {
+		exemplar := exemplars.At(i)
+		if !exemplar.TraceID().IsEmpty() && !exemplar.SpanID().IsEmpty() {
+			attrs = append(attrs, attributesToMap(exemplar.FilteredAttributes()))
+			times = append(times, exemplar.Timestamp().AsTime())
+			floatValues = append(floatValues, exemplar.DoubleValue())
+			intValues = append(intValues, exemplar.IntValue())
+			traceID := exemplar.TraceID()
+			traceIDs = append(traceIDs, hex.EncodeToString(traceID[:]))
+			spanID := exemplar.SpanID()
+			spanIDs = append(spanIDs, hex.EncodeToString(spanID[:]))
+		}
+	}
+	return attrs, times, floatValues, intValues, traceIDs, spanIDs
+}
+
+func attributesToMap(attributes pcommon.Map) map[string]string {
+	m := make(map[string]string, attributes.Len())
+	attributes.Range(func(k string, v pcommon.Value) bool {
+		m[k] = v.Str()
+		return true
+	})
+	return m
+}
+
+func convertSliceToArraySet(slice interface{}) clickhouse.ArraySet {
+	var set clickhouse.ArraySet
+	switch slice := slice.(type) {
+	case []uint64:
+		for _, item := range slice {
+			set = append(set, item)
+		}
+	case []float64:
+		for _, item := range slice {
+			set = append(set, item)
+		}
```

Comment:
do you want to catch a default to know something is missing?

Reply:
done